### ISSUE TITLE
When a moderator ends a meeting, indicate to viewers who ended the meeting

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/bigbluebutton#9856]
To reduce support issues, currently users may not realize that when they get kicked out of a meeting it was due to a moderator ending the meeting.

Add to the message shows to viewers that `The meeting was ended by <Moderator Name>`.

**Comment by gnaegi** [comment]
We just had a meeting with 100 user being ended accentually by one of the moderators who just wanted to leave. It would help to separate the exit and logout actions and to make it more difficult to actually end a meeting..

**Comment by svergata** [comment]
I welcome this message at meeting end. 
It just helping out in grep all clusterlogs for the causing User to get him notified. 
@gnaegi
There are seperatet action for that. one is a Question at ending a meeting "Do you want to end the meeting" yes/no. The other one leaves immediately and eventually presents a redirection with OK.

**Comment by gnaegi** [comment]
@svergata thanks your hint. Yes, I know, it is technically all correct. But in a hurry people press the wrong link and people just don't read those dialogs :-( 

From a usability standpoint I think the two actions are to close and the severity of choosing the wrong button and then actually kicking everybody out of a meeting, loosing all the notes you made for hours in the notes tool, loosing all the annotations on the slides is heavy.  "End the meeting" could also be interpreted as "end this meeting for me". 

In our system users constantly deleted things because the delete link was at a wrong place and the dialog was too simple and again, people just don't read the confirmation dialogs anymore. The user obviously thought he pressed "leave", so why reading the message that he is going to leave? He will just press yes. It is useless. 

E.g. we could have a dialog that requires more than just a button click when the meetings still has active webcams or mics which might indicate an unintentional ending. In our system we added a checkbox the user needs check for all severe actions to distinct from not harmfull actions. 

Another good idea is to list all the users that will be kicked out of the meeting to make the user aware of the impact of his action. In this case users will not us press the ok button because visually they realize that something is wrong. Why would I get a list of usernames when I just want to leave the meeting? So you get the desired attention that the dialog deserves.  

I also think that the "leave" action should not be hidden in the ... symbol in the first place, I think this is very rarely used. People just close the window, why bothering searching for a "leave" action when you can just close the window? I surely never use the link. The link could be placed outside the .... menu or added to the menu that appears on the avatar (where you can change your state etc). 

The message fix by @ffdixon is a great quick fix that helps to understand what happens, however it does not prevent it from happening unintentionally. 

I was not aware of this problem until our conference with 100 users got ended twice. I am sure we are not the only ones who are faced with this usability issue. I also think it is very difficult to anticipate such a usability problem during development, it seems perfectly reasonable to have the two links there and behave the way they behave. But production shows us that an adjustment might help reducing confusion and frustration a lot. 

Despite this issue: BBB is absolutely fabulous and I appreciate the hard work of the devs team!

**Comment by matiasilva** [comment]
I have mixed feelings about this. You must have a native way to log out of the meeting, just closing the window shouldn't be the recommended option. If, even after adding a confirmation screen, your users continue to press the wrong button, then that's on them now. Especially moderators, they should be more informed in the first place.

**Comment by gnaegi** [comment]
@matiasilva Thanks for your thoughts. I agree that actively leaving the meeting is better. I think the leave link is too hidden in the ... menu to be used by users and it is a click more than just closing the window. 

Regarding "moderators should be more informed": sure, i couldn't agree more :-) 

But still, it can happen also to experts to press the wrong button accidentially and I think listing all users that will be kicked out of the meeting would greatly improve the awareness of which action was triggered. Or a more alerty style dialog (e.g. in red) could also help.

**Comment by matiasilva** [comment]
@gnaegi I see. It often helps to have a visual reference, could you mock up a picture of what you would like to see for people to give some feedback? However, worth noting that the issue you're raising is separate from the issue being dealt with here originally. I would welcome you to open another issue for discussion.

**Comment by gnaegi** [comment]
@matiasilva thanks, I added a mockup in  #9868.

@ffdixon when we debugged what happened in our conference that was ended unluckely by someone I realized that the end event in the event.xml does not contain any information about who ended the meeting. It would be great if the user who ended does not only appear in the message in the meeting but is also in the events.xml  for the reason of transparency.
